packages/ibm-products/src/components/Datagrid/Datagrid/DatagridHeaderRow.js

```diff
@@ -44,6 +48,8 @@ const HeaderRow = (datagridState, headRef, headerGroup) => (
               header.getHeaderProps().className
             )}
             key={header.id}
+            title={header.Header.props?.title}
+            aria-hidden={!header.Header.props?.title}
```

Comment:
There is some nuance here because of how the structure of the column headers can change based on the hooks/plugins that are being used.

This is something that I've been looking at recently (particularly [here](https://github.com/carbon-design-system/ibm-products/pull/3344)). For example, if you add `useSortableColumns`, with this change all `th` elements will receive `aria-hidden="true"` which is not expected. This is because adding `useSortableColumns` changes `header.Header` (or `colDef.Header`) from just an object to a function.

Can we add a default title attribute to account for these instances? You can see this by looking at the `Sortable Columns` [story in the deploy preview](https://deploy-preview-3325--carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-components-datagrid-datagrid-canary--sortable-columns).

<img width="163" alt="image" src="https://github.com/carbon-design-system/ibm-products/assets/10215203/ac0b9c9a-dc29-458a-a5f1-b0a4cdd1d0de">